### PR TITLE
Add persistent volume for future audit log

### PR DESCRIPTION
**Pull request by coreydaley**
Add the creating and mounting of a persistent valume at /var/log/audit for the future audit logs to be written to using a default install

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [pre-commit](https://pre-commit.com/) utility can be used to generate the necessary content. Use `pre-commit run -a` to apply changes.
- [x] JSON Schema template updated and re-generated the raw schema via `pre-commit` hook.
- [ ] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

**Comment by coreydaley** [comment]
/assign @rm3l @gazarenkov 
This is to support the audit log functionality

File: charts/backstage/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  capacity:
+    storage: 5Gi
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ printf "%s-audit-log" .Release.Name }}
+spec:
```
Comment (rm3l) [inline charts/backstage/templates/pvc.yaml:15]: I think the storage class should at least be configurable from the `values.yaml` file. In https://issues.redhat.com/browse/RHIDP-2705, there was a request to make this configurable for the dynamic plugins volume in environments with no default storage class available.

File: charts/backstage/templates/pvc.yaml
Comment (rm3l) [inline charts/backstage/templates/pvc.yaml:1]: Currently, this PVC is deleted when the Helm Release is uninstalled. I wonder if we should not keep it (just like the DB PVC)..

File: charts/backstage/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: PersistentVolume
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply charts/backstage/templates/pvc.yaml:2]: PersistentVolume != PersistenVolumeClaim, they are two separate things.

File: charts/backstage/templates/pvc.yaml (hunk 1 above)
Comment (coreydaley) [reply charts/backstage/templates/pvc.yaml:15]: Can we leave that for another pull request and I will update both this one and the dynamic plugins volume?

File: charts/backstage/templates/pvc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rm3l) [reply charts/backstage/templates/pvc.yaml:2]: Ah I just realized that you were creating both a PV and a PVC, but there is no YAML separator in your `pvc.yaml` file.
But why would you need to also create a PV explicitly? That's the role of a PVC to dynamically provision a volume for you, no?

[the remark this answers is not in the document]
Comment (rm3l) [reply charts/backstage/templates/pvc.yaml:2]: I think both approaches are possible. A PV could be statically created by an admin, then a PVC can map to that PV. Or a PVC could also result in a PV provisioned dynamically.
I just tested this PR on a ClusterBot cluster by commenting out the PV resource (https://github.com/coreydaley/redhat-developer-rhdh-chart/blob/2024-07-15-audit-logging/charts/backstage/templates/pvc.yaml#L1-L10), and the PVC was successfully bound to the Deployment:
```
$ k get pvc              
NAME                                                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-my-backstage-1-postgresql-0                       Bound    pvc-f8a17372-ecba-4df2-9879-0d70adb58b04   1Gi        RWO            gp3            63m
my-backstage-1-66954c8b9b-xnbrc-dynamic-plugins-root   Bound    pvc-af674dcd-ec00-417e-a8d6-be9dd92ce163   2Gi        RWO            gp3            63m
my-backstage-1-audit-log                               Bound    pvc-6c4bc27f-be6f-4d62-aeaf-109c16a663be   5Gi        RWO            gp3            63m
```

I was asking about the need for creating a PV here, as that might not be allowed in some environments.

File: charts/backstage/templates/pvc.yaml
Comment (coreydaley) [reply charts/backstage/templates/pvc.yaml:1]: Thanks Kim!

File: charts/backstage/templates/pvc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (coreydaley) [reply charts/backstage/templates/pvc.yaml]: @rm3l 
Works on dev sandbox (your change does) so i have updated the pull request to remove the pv creation. ptal

**Review by rm3l — APPROVED** [review]

**Pull request merged**